src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -2993,11 +2993,162 @@ static void Main()
     }
 }
 ";
+            var comp = CompileAndVerify(source, expectedOutput: " 1 2", additionalRefs: new[] { ValueTupleRef, SystemRuntimeFacadeRef });
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TypeMergingSuccess2()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        (string x1, byte x2, var x3) = (null, 2, 3);
+        System.Console.WriteLine(x1 + "" "" + x2 + "" "" + x3);
+    }
+}
+";
+            var comp = CompileAndVerify(source, expectedOutput: " 2 3", additionalRefs: new[] { ValueTupleRef, SystemRuntimeFacadeRef });
```

Comment:
Need to add a semantic info test for this scenario. We should see that the type of the `(null, 2, 3)` is `null`, converted type is `(string, byte, int)` and conversion is ImplicitTupleLiteral